utils/histogram.hh

```diff
@@ -258,7 +258,17 @@ public:
         rate_moving_average res;
         if (_count > 0) {
```

Comment:
Typically, _count will be greater than zero, you can move the duration calculation outside of the if, and add elapsed > 1.0 (or any other value for that matter) to be part of the if condition.

Reply:
all in all, it looks good and cool use of C++20 attributes.

Reply:
@amnonh 
Just to see that we are on the same page, do you think that returning 0 until we have an interval of at least one second or any finite an reasonably large duration (0.5,0.25...) is an acceptable solution? (this will ensure that _count/elapse will never be infinite). If so , unfortunately I will have to drop the cool c++ 20 feature or to add a [[likely]]
attribute to the new consolidation condition :slightly_smiling_face:  but do you agree to that solution?